test/services/javascript_scorer_test.rb

```diff
@@ -0,0 +1,216 @@
+      ]
+
+      # Can be in any order
+      best_docs = [
+        { id: 2, rating: 1 },
+        { id: 1, rating: 0 },
+        { id: 3, rating: 1 }
+      ]
+
+      score = javascript_scorer.score(docs, best_docs, scorer_code)
+      assert_equal 0.25, score
+    end
+
+    test 'situation produces NaN' do
+      # Need David Fisher help here.  Why am I getting NaN?
+      # Going to just make it return a 0 in FetchService
+
+      docs = [ { :id => '77383738', :rating => 0.0 },
+               { :id => '77502729', :rating => 0.0 },
+               { :id => '77031393', :rating => 0.0 },
+               { :id => '78106266', :rating => 0.0 } ]
+
+      best_docs = [ { :id => '77193049', :rating => 0.0 },
+                    { :id => '77031393', :rating => 0.0 },
+                    { :id => '2120998', :rating => 0.0 } ]
+
+      score = javascript_scorer.score(docs, best_docs, scorer_code)
+      assert score.nan?
+    end
+  end
+
+  describe 'cg@10' do
+    let(:the_case) { cases(:case_without_score) }
+
+    test 'runs simple' do
+      scorer_code = File.read('db/scorers/cg@10.js')
+
+      # order matters!
+      docs = [
+        { id: 1, rating: 0 },
+        { id: 2, rating: 3 },
+        { id: 3, rating: 1 }
+      ]
+
+      best_docs = []
+
+      score = javascript_scorer.score(docs, best_docs, scorer_code)
+      assert_equal 4, score
+    end
+  end
+
+  describe 'dcg@10' do
+    let(:the_case) { cases(:case_without_score) }
+
+    test 'runs simple' do
+      scorer_code = File.read('db/scorers/dcg@10.js')
+
+      # order matters!
+      docs = [
+        { id: 1, rating: 0 },
+        { id: 2, rating: 3 },
+        { id: 3, rating: 1 }
+      ]
+
+      best_docs = []
+
+      score = javascript_scorer.score(docs, best_docs, scorer_code)
+      assert_equal 4.92, score
+    end
+  end
+
+  describe 'ndcg@10' do
+    let(:the_case) { cases(:case_without_score) }
+
+    test 'runs simple' do
+      scorer_code = File.read('db/scorers/ndcg@10.js')
+
+      # order matters!
+      docs = [
+        { id: 1, rating: 0 },
+        { id: 2, rating: 3 },
+        { id: 3, rating: 1 }
+      ]
+
+      # order matters!
+      best_docs = [
+        { id: 2, rating: 3 },
+        { id: 3, rating: 1 },
+        { id: 1, rating: 0 }
+      ]
+
+      score = javascript_scorer.score(docs, best_docs, scorer_code)
+      assert_equal 0.64, score
+    end
+  end
+  describe 'rr@10' do
+    let(:the_case) { cases(:case_without_score) }
+
+    test 'runs simple' do
+      scorer_code = File.read('db/scorers/rr@10.js')
+
+      # order matters!
+      docs = [
+        { id: 1, rating: 0 },
+        { id: 2, rating: 0 },
+        { id: 3, rating: 1 }
+      ]
+
+      best_docs = []
+
+      score = javascript_scorer.score(docs, best_docs, scorer_code)
+      assert_equal 0.33, score
+    end
+  end
+
+  describe 'recency scorer' do
+    let(:the_case) { cases(:case_without_score) }
+
+    test 'runs simple and tests eachDoc w/ a function' do
```

Comment:
I'd love to get a bit more detail on how to pick a baseDate..  and if the baseDate could be "today" instead?  @david-fisher